### PR TITLE
HOCS-2234: add capability to save/update/delete somu item to elastic search

### PR DESCRIPTION
**Pull request by LiamMacP**
This pull request adds the necessary routing for the elastic search to allow for the addings, removing and deleting of somu items to the elastic search index. This works by adding 3 different routes through the search consumer that will parse the audit input and direct based on the event type to the correct queue. 

This then updates the case data object and does the required action on the set within the case data object.

Tests have also been added to check functionality is as expected.

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/hocs/search/queue/SearchConsumerTest.java [hunk 1]
```diff
@@ -142,6 +180,16 @@ public void shouldCallDeleteTopic() throws JsonProcessingException {
...
 
+    @Test
+    public void shouldCallDeleteTopic() throws JsonProcessingException {
+
+        CreateAuditDto auditDto = new CreateAuditDto(caseUUID, "\"11111111-2222-2222-2222-333333333333\"", EventType.CASE_TOPIC_DELETED.toString());
+        String json = mapper.writeValueAsString(auditDto);
+        template.sendBody(searchQueue, json);
+        verify(mockDataService, times(1)).deleteTopic(eq(caseUUID), eq("11111111-2222-2222-2222-333333333333"));
+        verifyNoMoreInteractions(mockDataService);
+    }*/
```
Comment (zuzak) [inline src/test/java/uk/gov/digital/ho/hocs/search/queue/SearchConsumerTest.java:191]: Why are these commented out?

Comment (LiamMacP) [reply src/test/java/uk/gov/digital/ho/hocs/search/queue/SearchConsumerTest.java:191]: Very good question, they are copied and pasted from above for me to work off. Can be removed as the test is still present and I didn't use it.

Comment (LiamMacP) [reply src/test/java/uk/gov/digital/ho/hocs/search/queue/SearchConsumerTest.java]: This has been rectified 👍🏼

File: src/main/java/uk/gov/digital/ho/hocs/search/application/queue/SearchConsumer.java [hunk 2]
```diff
@@ -101,6 +105,15 @@ public void configure() {
...
                 .to(DELETE_TOPIC_QUEUE)
                 .endChoice()
+                .when(simple("${property.type} == '" + EventType.SOMU_ITEM_CREATED + "'"))
+                .to(CREATE_SOMU_ITEM_QUEUE)
+                .endChoice()
+                .when(simple("${property.type} == '" + EventType.SOMU_ITEM_DELETED + "'"))
+                .to(DELETE_SOMU_ITEM_QUEUE)
+                .endChoice()
+                .when(simple("${property.type} == '" + EventType.SOMU_ITEM_UPDATED + "'"))
+                .to(UPDATE_SOMU_ITEM_QUEUE)
```
Comment (parth-ho) [inline src/main/java/uk/gov/digital/ho/hocs/search/application/queue/SearchConsumer.java:115]: Not for now, but we might wanna think about refactoring this to as it's getting quite long. Also, I reckon we'll be adding to this consumer a lot more in the future.

**Comment by parth-ho** [comment]
Looks good to me!

**Review by parth-ho — APPROVED** [review]

File: src/main/java/uk/gov/digital/ho/hocs/search/application/queue/SearchConsumer.java (hunk 2 above)
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/search/application/queue/SearchConsumer.java:115]: Yeah I was thinking that, needs some work on best way to do it 👍🏼

**Pull request merged**